catalog/plugin/catalog-plugin-metadatabackup/src/main/java/org/codice/ddf/catalog/plugin/metadata/backup/MetadataBackupPlugin.java

```diff
@@ -0,0 +1,203 @@
+        return folderDepth;
+    }
+
+    public void setFolderDepth(Integer folderDepth) {
+        folderDepth = Math.max(folderDepth, MIN_FOLDER_DEPTH);
+        folderDepth = Math.min(folderDepth, MAX_FOLDER_DEPTH);
+        this.folderDepth = folderDepth;
+    }
+
+    private void processRequest(ProcessRequest<? extends ProcessResourceItem> processRequest)
+            throws PluginExecutionException {
+        LOGGER.trace("Backing up metadata");
+        if (StringUtils.isEmpty(outputDirectory)) {
+            throw new PluginExecutionException(
+                    "Unable to backup ingested metadata; no outputDirectory.");
+        }
+
+        List<? extends ProcessResourceItem> processResourceItems = processRequest.getProcessItems();
+        for (ProcessResourceItem processResourceItem : processResourceItems) {
+            Metacard metacard = processResourceItem.getMetacard();
+            Attribute attribute = metacard.getAttribute(Core.METADATA);
+            if (attribute != null) {
+                Serializable value = attribute.getValue();
+                if (value instanceof String) {
+                    LOGGER.debug("Backing up metadata for metacard : {}", metacard.getId());
+                    copyMetadataToOutputDirectory((String) value, metacard.getId());
+                }
+            }
+        }
+    }
+
+    private void copyMetadataToOutputDirectory(String metadata, String metacardId) {
+        File parent = getCompleteDirectory(getFolderDepth(), metacardId);
+        if (!parent.exists()) {
+            try {
+                FileUtils.forceMkdir(parent);
+            } catch (IOException e) {
+                LOGGER.debug("Unable to make directory {}", parent.getPath(), e);
+            }
+        }
+
+        validateDirectory(parent);
+        File file = new File(parent.getPath() + File.separator + metacardId + ".xml");
+        LOGGER.trace("Writing metadata from {} to file {}", metacardId, file.getPath());
+
+        try {
+            FileUtils.writeStringToFile(file, metadata);
+        } catch (IOException e) {
+            LOGGER.debug("Unable to backup {}. ", metacardId, e);
+        }
+    }
+
+    private void deleteMetadataIfPresent(String filename) {
+        File parent = getCompleteDirectory(getFolderDepth(), filename);
+        validateDirectory(parent);
+        File file = new File(parent.getPath() + File.separator + filename + ".xml");
+        LOGGER.trace("Deleting metadata file {}", file.getPath());
+        try {
+            Files.deleteIfExists(file.toPath());
+        } catch (IOException e) {
+            LOGGER.debug("Unable to backup metadata {}. ", filename, e);
+        }
+        clearEmptyDirectories(parent);
+    }
+
+    private File getCompleteDirectory(int depth, String id) {
+        File parent = new File(getOutputDirectory());
+        for (int i = 0; i < depth && id.length() > (i * 2 + 2); i++) {
+            parent = new File(parent, id.substring(i * 2, i * 2 + 2));
```

Comment:
I think this is creating directories based off of the ID 2 characters at a time. I'm a little confused on the +2? If it's b/c we are starting the loop at 0, would it make it cleaner to just start with `i = 1`?
Also, if the depth is configurable should we change the directory logic to try to more evenly divide the id, rather than going 2 characters at a time until the depth is reached?